catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/ListApplication.java

```diff
@@ -0,0 +1,369 @@
+import ddf.catalog.content.operation.impl.CreateStorageRequestImpl;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.operation.CreateResponse;
+import ddf.catalog.source.IngestException;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.mime.MimeTypeMapper;
+import ddf.mime.MimeTypeResolutionException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.activation.MimeType;
+import javax.activation.MimeTypeParseException;
+import javax.servlet.MultipartConfigElement;
+import javax.servlet.http.Part;
+import javax.ws.rs.InternalServerErrorException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response.Status;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.platform.util.uuidgenerator.UuidGenerator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.Response;
+import spark.servlet.SparkApplication;
+
+public class ListApplication implements SparkApplication {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ListApplication.class);
+
+  private static final String LIST_TYPE_HEADER = "List-Type";
+
+  private static final String DEFAULT_FILE_EXTENSION = "bin";
+
+  private static final String DEFAULT_MIME_TYPE = "application/octet-stream";
+
+  private static final String DEFAULT_FILE_NAME = "file";
+
+  private static final List<String> REFINEABLE_MIME_TYPES =
+      Arrays.asList(DEFAULT_MIME_TYPE, "text/plain");
+
+  private final MimeTypeMapper mimeTypeMapper;
+
+  private final CatalogFramework catalogFramework;
+
+  private final UuidGenerator uuidGenerator;
+
+  private final SplitterLocator splitterLocator;
+
+  public ListApplication(
+      MimeTypeMapper mimeTypeMapper,
+      CatalogFramework catalogFramework,
+      UuidGenerator uuidGenerator,
+      SplitterLocator splitterLocator) {
+    this.mimeTypeMapper = mimeTypeMapper;
+    this.catalogFramework = catalogFramework;
+    this.uuidGenerator = uuidGenerator;
+    this.splitterLocator = splitterLocator;
+  }
+
+  @Override
+  public void init() {
+    post(
+        "/list/import",
+        (request, response) -> {
+          MultipartConfigElement multipartConfigElement = new MultipartConfigElement("");
+          request.raw().setAttribute("org.eclipse.jetty.multipartConfig", multipartConfigElement);
+
+          String listType = request.headers(LIST_TYPE_HEADER);
+
+          if (StringUtils.isBlank(listType)) {
+            String exceptionMessage = String.format("The header %s must be set.", LIST_TYPE_HEADER);
+            LOGGER.info(exceptionMessage);
```

Comment:
Should this be at the debug level?

Reply:
Normally, I would say `yes`. But this is consistent with the way RESTEndpoint does it when it ingests content.